cedar-policy-core/src/entities/conformance.rs

```diff
@@ -190,7 +253,7 @@ pub fn typecheck_restricted_expr_against_schematype(
     // directly?
```

Comment:
comment can be updated/removed.  (And this issue can also be `resolves #440`)

Reply:
I'm hoping John comes up with a way to avoid calling `schematype_of_restricted_expr` without impacting error reporting (maybe only calling if if the types don't match). I wasn't sure if we want to call 440 resolved while we're still using `schematype_of_restricted_expr` on the main path.

Reply:
Done